includes/classes/class-plugin-manager.php

```diff
@@ -0,0 +1,409 @@
+							}
+							?>
+						</div>
+					</div>
+				</form>
+			</div>
+			<div class="clear"></div>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render individual plugin cards.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $standalone_plugin Plugin slug as passed from get_standalone_plugins().
+	 *
+	 * @return void
+	 */
+	private static function render_plugin_card( string $standalone_plugin ) {
+		$plugin = plugins_api(
+			'plugin_information',
+			array(
+				'slug'   => $standalone_plugin,
+				'fields' => array(
+					'short_description' => true,
+					'icons'             => true,
+				),
+			)
+		);
+		if ( is_object( $plugin ) ) {
+			$plugin = (array) $plugin;
+		}
+
+		// Remove any HTML from the description.
+		$description = wp_strip_all_tags( $plugin['short_description'] );
+		$title       = $plugin['name'];
+
+		/**
+		 * Filters the plugin card description on the Add Plugins screen.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param string $description Plugin card description.
+		 * @param array  $plugin      An array of plugin data. See {@see plugins_api()}
+		 *                            for the list of possible values.
+		 */
+		$description = apply_filters( 'plugin_install_description', $description, $plugin );
+		$version     = $plugin['version'];
+		$name        = wp_strip_all_tags( $title . ' ' . $version );
+		$author      = $plugin['author'];
+		if ( ! empty( $author ) ) {
+			/* translators: %s: Plugin author. */
+			$author = ' <cite>' . sprintf( __( 'By %s', 'default' ), $author ) . '</cite>';
+		}
+
+		$requires_php = isset( $plugin['requires_php'] ) ? $plugin['requires_php'] : null;
+		$requires_wp  = isset( $plugin['requires'] ) ? $plugin['requires'] : null;
+
+		$compatible_php = is_php_version_compatible( $requires_php );
+		$compatible_wp  = is_wp_version_compatible( $requires_wp );
+		$tested_wp      = ( empty( $plugin['tested'] ) || version_compare( get_bloginfo( 'version' ), $plugin['tested'], '<=' ) );
+		$action_links   = array();
+
+		if ( current_user_can( 'install_plugins' ) || current_user_can( 'update_plugins' ) ) {
+			$status = install_plugin_install_status( $plugin );
+
+			switch ( $status['status'] ) {
+				case 'install':
+					if ( $status['url'] ) {
+						if ( $compatible_php && $compatible_wp ) {
+							$action_links[] = sprintf(
+								'<a class="install-now button" data-slug="%s" href="%s" aria-label="%s" data-name="%s">%s</a>',
+								esc_attr( $plugin['slug'] ),
+								esc_url( $status['url'] ),
+								/* translators: %s: Plugin name and version. */
+								esc_attr( sprintf( _x( 'Install %s now', 'plugin', 'default' ), $name ) ),
+								esc_attr( $name ),
+								__( 'Install Now', 'default' )
+							);
+						} else {
+							$action_links[] = sprintf(
+								'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+								_x( 'Cannot Install', 'plugin', 'default' )
+							);
+						}
+					}
+					break;
+
+				case 'update_available':
```

Comment:
I thought we were not supporting updating in this screen because there were issues with it?

Reply:
Only with updating via ajax were there some quirks, bit without it should work fine according to Slack

Reply:
@felixarntz, @swissspidy is correct. When updating via Ajax there are a lot of pre/post-requisites that the code fails on. 

Removing the ajax dependency works, but updating sends the user to the core screen and they do not return to the wpp setting screen. 

I'd not yet found a reliable means around this, but I'll take another crack at it.

Reply:
I'd argue it's not important to offer updating on this page, it's not something we're trying to achieve, and just a nice to have.

Given the subpar experience, maybe we just remove the update UI on this page entirely? It's not important here, and just a bit of an extra burden we'll have to maintain. Given most of this is duplicated UI from core, I'd try to limit it to the essential. So unless removing the ability to update here introduces other `updates.js` interoperability problems I can't anticipate right now, I'd say we should remove this UI state. See also https://github.com/WordPress/performance/pull/864#discussion_r1380803664.

WDYT @10upsimon @swissspidy?